File: Services/Personnel/Personnel.Domain/AggregateModel/JobPostingAggregate/JobPosting.cs
using System;
using System.Collections.Generic;
using System.Text;
using Personnel.Domain.Common;
using Personnel.Domain.Events;

namespace Personnel.Domain.AggregateModel.JobPostingAggregate
{
    public class JobPosting: Entity, IAggregateRoot
    {
        public JobTitle JobTitle { get; }
        public string Description { get; }
        public DateTime PostedTime { get; } = DateTime.Now;

        public JobPosting(JobTitle jobTitle, string description)
        {
            JobTitle = jobTitle;
            Description = description;

            AddDomainEvent(new JobPostingCreatedDomainEvent(this));
        }
    }
}

File: WebApps/AngularSPA/Client/src/app/personnel/models/JobPosting.ts
import { schema } from 'normalizr';
import { jobTitleSchema } from './JobTitle';

export const jobPostingSchema = new schema.Entity('jobPostings', {
    jobTitle: jobTitleSchema
});

export const jobPostingListSchema = new schema.Array(jobPostingSchema);

export interface JobPosting {
    id: string,
    jobTitle: string,
    description: string,
    postedTime: Date
}